marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -269,6 +285,23 @@ function getPropertyValues(currentProperty, entityInstance) {
   return resultObject;
 }
 
+// returns null to use uri
+function getPrimaryValue(entityInstance, entityDefinition) {
+  let primaryKeyData = {}
+  if (entityDefinition.hasOwnProperty("primaryKey")) {
+    let primaryKey = entityDefinition.primaryKey;
+    if (entityInstance.hasOwnProperty((primaryKey))) {
+      primaryKeyData.propertyPath = primaryKey;
+      primaryKeyData.propertyValue = entityInstance[primaryKey];
```

Comment:
I think there's one more scenario to support here - the primaryKey property exists, but it does not have a value. Would be unusual, but could happen. So if entityInstance[primaryKey] is null or an empty string, return null here as well.